sdk/eventhubs/azure_messaging_eventhubs/examples/eventhubs_properties.rs

```diff
@@ -18,22 +17,17 @@ async fn main() -> Result<()> {
 
     let credential = DefaultAzureCredential::new()?;
 
-    let client = ProducerClient::new(
-        host,
-        eventhub.clone(),
-        credential.clone(),
-        Some(ProducerClientOptions {
-            application_id: Some("test_get_properties".to_string()),
-            ..Default::default()
-        }),
-    );
-    let result = client.open().await;
-    info!("Open result: {:?}", result);
-    if result.is_err() {
-        println!("Error opening client: {:?}", result.err());
+    let result = ProducerClient::builder(host.as_str(), eventhub.as_str(), credential.clone())
+        .with_application_id("test_get_properties")
+        .open()
+        .await;
+    if let Err(err) = result {
+        println!("Error opening client: {:?}", err);
         return Ok(());
     }
```

Comment:
That seems pretty fatal. You're already set up to use `?` so why not just do that and let the example program fail with a more useful message? Also, all `std::error::Error`s impl `Display` (it's a constraint) so you don't need to use the `Debug` notation to format. Generally will be more user-friendly, as is `Display` meant to be over `Debug`.

Reply:
In my experience, ? doesn't give a great error - all it does is print the error text - I'd lose the context for the error.

Reply:
You won't get a stacktrace this way either.